### PR TITLE
Handle gracefully a bad module 'results' key

### LINKED ISSUE #53962: Unexpected Exception, this is probably a bug: argument of type 'int' is not iterable
##### SUMMARY

When I create a custom module, return a list of integers, and name the return variable "results", I get the following error:
> ERROR! Unexpected Exception, this is probably a bug: argument of type 'int' is not iterable

If I change the return value to anything other than "results", it works.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

Custom modules

##### ANSIBLE VERSION

```paste below
ansible 2.7.5
  config file = /home/matt/ansible_playbooks/ansible.cfg
  configured module search path = ['/home/matt/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.6/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 3.6.7 (default, Dec  5 2018, 15:02:05) [GCC 4.8.5 20150623 (Red Hat 4.8.5-36)]
```

##### CONFIGURATION

```paste below
DEPRECATION_WARNINGS(/home/matt/ansible_playbooks/ansible.cfg) = False
HOST_KEY_CHECKING(/home/matt/ansible_playbooks/ansible.cfg) = False
PERSISTENT_COMMAND_TIMEOUT(/home/matt/ansible_playbooks/ansible.cfg) = 60
```

##### OS / ENVIRONMENT

OS: CentOS Linux 7 (Core)

##### STEPS TO REPRODUCE

Create the following custom Python module in the "library" folder.  It can be named anything.  Here it is "test_lib.py"
```python
#!/usr/bin/python3

from ansible.module_utils.basic import AnsibleModule


def main():
    """Main function."""

    module = AnsibleModule(
        argument_spec=dict(
        )
    )

    return_list = [1, 2, 3, 4, 5]
    module.exit_json(results=return_list)


if __name__ == "__main__":
    main()
```

Create the following test playbook in the root ansible playbooks directory:
```yaml
---
- name: Demonstrate bug
  hosts: localhost
  gather_facts: false
  vars:
    ansible_python_interpreter: ./venv/bin/python3

  tasks:
    - name: Test output
      test_lib:
      register: numbers

    - debug:
        var: numbers
```

Run the playbook with "ansible-playbook <pb-name>.yml"

##### EXPECTED RESULTS

The playbook should print out the list [1, 2, 3, 4, 5]

##### ACTUAL RESULTS

It throws the following error on the first task:

> ERROR! Unexpected Exception, this is probably a bug: argument of type 'int' is not iterable
> to see the full traceback, use -vvv

```paste below
ansible-playbook 2.7.5
  config file = /home/matt/ansible_playbooks/ansible.cfg
  configured module search path = ['/home/matt/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.6/site-packages/ansible
  executable location = /usr/bin/ansible-playbook
  python version = 3.6.7 (default, Dec  5 2018, 15:02:05) [GCC 4.8.5 20150623 (Red Hat 4.8.5-36)]
Using /home/matt/ansible_playbooks/ansible.cfg as config file
setting up inventory plugins
/etc/ansible/hosts did not meet host_list requirements, check plugin documentation if this is unexpected
/etc/ansible/hosts did not meet script requirements, check plugin documentation if this is unexpected
Parsed /etc/ansible/hosts inventory source with ini plugin
 [WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'

Loading callback plugin default of type stdout, v2.0 from /usr/lib/python3.6/site-packages/ansible/plugins/callback/default.py

PLAYBOOK: test.yml ***********************************************************************************************************************************************************************************************************************************************************
1 plays in test.yml

PLAY [Demonstrate bug] *******************************************************************************************************************************************************************************************************************************************************
META: ran handlers

TASK [Test output] ***********************************************************************************************************************************************************************************************************************************************************
task path: /home/matt/ansible_playbooks/test.yml:9
<127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: matt
<127.0.0.1> EXEC /bin/sh -c 'echo ~matt && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/matt/.ansible/tmp/ansible-tmp-1552918894.7886205-102956225615398 `" && echo ansible-tmp-1552918894.7886205-102956225615398="` echo /home/matt/.ansible/tmp/ansible-tmp-1552918894.7886205-102956225615398 `" ) && sleep 0'
Using module file /home/matt/ansible_playbooks/library/test_lib.py
<127.0.0.1> PUT /home/matt/.ansible/tmp/ansible-local-28981nnl4ulrc/tmp8wf8hk63 TO /home/matt/.ansible/tmp/ansible-tmp-1552918894.7886205-102956225615398/AnsiballZ_test_lib.py
<127.0.0.1> EXEC /bin/sh -c 'chmod u+x /home/matt/.ansible/tmp/ansible-tmp-1552918894.7886205-102956225615398/ /home/matt/.ansible/tmp/ansible-tmp-1552918894.7886205-102956225615398/AnsiballZ_test_lib.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c './venv/bin/python3 /home/matt/.ansible/tmp/ansible-tmp-1552918894.7886205-102956225615398/AnsiballZ_test_lib.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c 'rm -f -r /home/matt/.ansible/tmp/ansible-tmp-1552918894.7886205-102956225615398/ > /dev/null 2>&1 && sleep 0'
ERROR! Unexpected Exception, this is probably a bug: argument of type 'int' is not iterable
the full traceback was:

Traceback (most recent call last):
  File "/usr/bin/ansible-playbook", line 118, in <module>
    exit_code = cli.run()
  File "/usr/lib/python3.6/site-packages/ansible/cli/playbook.py", line 122, in run
    results = pbex.run()
  File "/usr/lib/python3.6/site-packages/ansible/executor/playbook_executor.py", line 156, in run
    result = self._tqm.run(play=play)
  File "/usr/lib/python3.6/site-packages/ansible/executor/task_queue_manager.py", line 291, in run
    play_return = strategy.run(iterator, play_context)
  File "/usr/lib/python3.6/site-packages/ansible/plugins/strategy/linear.py", line 325, in run
    results += self._wait_on_pending_results(iterator)
  File "/usr/lib/python3.6/site-packages/ansible/plugins/strategy/__init__.py", line 712, in _wait_on_pending_results
    results = self._process_pending_results(iterator)
  File "/usr/lib/python3.6/site-packages/ansible/plugins/strategy/__init__.py", line 117, in inner
    results = func(self, iterator, one_pass=one_pass, max_passes=max_passes)
  File "/usr/lib/python3.6/site-packages/ansible/plugins/strategy/__init__.py", line 474, in _process_pending_results
    if task_result.is_failed():
  File "/usr/lib/python3.6/site-packages/ansible/executor/task_result.py", line 59, in is_failed
    'results' in self._result and True in [True for x in self._result['results'] if 'failed_when_result' in x]:
  File "/usr/lib/python3.6/site-packages/ansible/executor/task_result.py", line 59, in <listcomp>
    'results' in self._result and True in [True for x in self._result['results'] if 'failed_when_result' in x]:
TypeError: argument of type 'int' is not iterable
```

### PR DESCRIPTION
**Pull request by bcoca**
fixes #53962

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

core

### CODE REVIEW

File: lib/ansible/plugins/action/__init__.py [hunk 1]
```diff
@@ -909,6 +910,12 @@ def _execute_module(self, module_name=None, module_args=None, tmp=None, task_var
         if data.pop("_ansible_suppress_tmpdir_delete", False):
             self._cleanup_remote_tmp = False
 
+        # NOTE: yum returns results .. but that made it 'compatible' with squashing, so we allow mappings, for now
+        if 'results' in data and not isinstance(data['results'], Mapping):
```
Comment (mkrizek) [inline lib/ansible/plugins/action/__init__.py]: IIRC `results` in the `yum` module is a `list`.

Comment (bcoca) [reply lib/ansible/plugins/action/__init__.py]: actually, `results` should always be a list .. of dicts, so this was wrong

File: lib/ansible/plugins/action/__init__.py [hunk 2]
```diff
@@ -909,6 +910,12 @@ def _execute_module(self, module_name=None, module_args=None, tmp=None, task_var
         if data.pop("_ansible_suppress_tmpdir_delete", False):
             self._cleanup_remote_tmp = False
 
+        # NOTE: yum returns results .. but that made it 'compatible' with squashing, so we allow mappings, for now
```
Comment (mkrizek) [inline lib/ansible/plugins/action/__init__.py:913]: The comment is now outdated.

**Review by jamescassell — APPROVED** [review]
shipit

**Pull request merged**